x-pack/solutions/search/packages/kbn-search-index-documents/lib/fetch_search_results.ts

```diff
@@ -19,9 +19,9 @@ export const fetchSearchResults = async (
   size: number = DEFAULT_DOCS_PER_PAGE,
   trackTotalHits: boolean = false
 ): Promise<Paginate<SearchHit>> => {
-  const result = await fetchWithPagination(
-    async () =>
-      await client.search({
+  return fetchWithPagination(
+    () =>
+      client.search({
```

Comment:
since `client.search()` already returns a promise both the `async` & `await` usage here is not necessary.